uint32 size
Data[] data
---
